File: lib/code_ocean/chat_gpt_request.rb
require 'net/http'
require 'uri'
require 'json'

class ChatGptRequest
  def request_gpt(prompt)
    url = URI.parse('https://api.openai.com/v1/chat/completions')
    data = {
      model: 'gpt-3.5-turbo',
      messages: [{ role: 'assistant', content: prompt }]
    }
    api_key = 'your api key'
    http = Net::HTTP.new(url.host, url.port)
    http.use_ssl = true
    request = Net::HTTP::Post.new(url.path, {'Content-Type' => 'application/json',
                                             'Authorization' => "Bearer #{api_key}"})
    request.body = data.to_json

    response = http.request(request)
    json_response = JSON.parse(response.body)
    content = json_response['choices'][0]['message']['content']

    return content
  end
end

File: lib/demo_python_excersise/chat_gpt_request.py
from openai import OpenAI

client = OpenAI(api_key='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')

# Set your OpenAI GPT-3 API key
def get_completion(prompt):
    response = client.chat.completions.create(
        messages=[
            {
                "role": "assistant",
                "content": prompt,
            }
        ],
        model="gpt-3.5-turbo",
    )

    # Extract the generated response from the OpenAI API response
    chat_response = response.choices[0].message.content
    return chat_response
